cartography/intel/crxcavator/crxcavator.py

```diff
@@ -0,0 +1,220 @@
+    :param update_tag: Timestamp used to determine data freshness
+    :return: None
+    """
+    ingestion_cypher = """
+    UNWIND {ExtensionsData} as extension
+    MERGE (e:ChromeExtension{id: extension.id})
+    ON CREATE SET
+    e.extension_id = extension.extension_id,
+    e.version = extension.version,
+    e.firstseen = timestamp()
+    SET
+    e.extcalls = extension.extcalls,
+    e.risk_total = extension.risk_total,
+    e.risk_metadata = extension.risk_metadata,
+    e.address = extension.address,
+    e.email = extension.email,
+    e.icon = extension.icon,
+    e.last_updated = extension.last_updated,
+    e.name = extension.name,
+    e.offered_by = extension.offered_by,
+    e.permissions_warnings = extension.permissions_warnings,
+    e.privacy_policy = extension.privacy_policy,
+    e.rating = extension.rating,
+    e.rating_users = extension.rating_users,
+    e.short_description = extension.short_description,
+    e.size = extension.size,
+    e.support_site = extension.support_site,
+    e.users = extension.users,
+    e.website = extension.website,
+    e.type = extension.type,
+    e.price = extension.price,
+    e.offline_enabled = extension.offline_enabled,
+    e.lastupdated = {UpdateTag}
+    """
+
+    logger.info('Ingesting {} extensions'.format(len(extensions)))
+    session.run(ingestion_cypher, ExtensionsData=extensions, UpdateTag=update_tag)
+
+
+def transform_user_extensions(user_extension_json):
+    """
+    Transforms the raw extensions JSON from the API into a list of extensions mapped to users
+    :param user_extension_json:  The JSON text blob returned from the CRXcavator API
+    :return: Tuple containing unique users list and extension info for ingestion
+    """
+    user_extensions = user_extension_json.items()
+    users_set = set()
+    extensions_by_user = []
+    for extension in user_extensions:
+        for details in extension[1].items():
+            for user in details[1]['users']:
+                users_set.add(user)
+                extensions_by_user.append({
+                    'id': "{0}|{1}".format(extension[0], details[0]),
+                    'user': user})
+    if len(users_set) == 0:
+        raise ValueError('No users returned from CRXcavator')
+    if len(extensions_by_user) == 0:
+        raise ValueError('No user->extension mapping returned from CRXcavator')
+
+    return list(users_set), extensions_by_user
+
+
+def load_user_extensions(users, extensions_by_user, session, update_tag):
+    """
+    Ingests the extension to user mapping details into Neo4J
+    :param users: List of user objects to create for mapping
+    :param extensions_by_user: List of user to extension id mappings
+    :param session: Neo4J session object for server communication
+    :param update_tag: Timestamp used to determine data freshness
+    :return: None
+    """
+
+    # todo: once G Suite ingestion is created, move first MERGE to that process
```

Comment:
I think MERGE is fine now and in the future. I think it's fair to assume that each intel modules aren't aware of others

Reply:
Based on discussion during design for this module. We're only creating this information here until full G Suite ingest is created. At that time, this would change to a relationship creation only.